pkg/operator/ceph/server.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type AdmissionController struct {
+	providerName    string
+	pathResourceMap map[string]webhook.Validator
+	certDir         string
+	port            int
+	scheme          *runtime.Scheme
+}
+
+func NewAdmissionController(providerName string, port int, certDir string, pathResourceMap map[string]webhook.Validator, scheme *runtime.Scheme) *AdmissionController {
+	return &AdmissionController{
+		providerName:    providerName,
+		port:            port,
+		certDir:         certDir,
+		pathResourceMap: pathResourceMap,
+		scheme:          scheme,
+	}
+}
+
+// StartServer will start the server
+func (a *AdmissionController) StartServer() {
```

Comment:
StartServer() should return an err so that the cmd package can call TerminateFatal() if there is some issue starting the webhook.
```suggestion
func (a *AdmissionController) StartServer() error {
```

Reply:
done